src/code/z_camera.c

```diff
@@ -526,282 +869,7416 @@ s32 Camera_GetWaterBoxBgCamSetting(Camera* camera, f32* waterY) {
+            camEyeCollision->sphNorm.yaw = diffSph->yaw;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeBATT0.s")
+        if (checkEye & 1) {
+            memcpy(&camCollision, camAtCollision, sizeof(CameraCollision));
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeKEEP1.s")
+        eyeAtBgId = Camera_BgCheckInfo(camera, &camera->eye, camAtCollision);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeKEEP2.s")
+        if (eyeAtBgId == 0) {
+            if (checkEye & 1) {
+                memcpy(camAtCollision, &camCollision, sizeof(CameraCollision));
+            } else {
+                return 3;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeKEEP3.s")
+        if (camEyeCollision->poly == camAtCollision->poly) {
+            return 3;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeKEEP4.s")
+        OLib_Vec3fToVecSphGeo(&camAtCollision->sphNorm, &camAtCollision->norm);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeKEEP0.s")
+        if (camAtCollision->sphNorm.pitch > 0x2EE0) { // 65.9 degrees
+            camAtCollision->sphNorm.yaw = BINANG_ROT180(diffSph->yaw);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeFIXD1.s")
+        if (atEyeBgId != eyeAtBgId) {
+            ret = 3;
+        } else {
+            cosEyeAt = Math3D_Parallel(&camEyeCollision->norm, &camAtCollision->norm);
+            if (cosEyeAt < -0.5f) {
+                ret = 6;
+            } else if ((cosEyeAt > 0.5f) || (checkEye & 2)) {
+                ret = 3;
+            } else {
+                ret = 1;
+            }
+        }
+    }
+    return ret;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeFIXD2.s")
+/**
+ * Calculates how much to adjust the camera at's y value when on a slope.
+ */
+f32 Camera_CalcSlopeYAdj(Vec3f* floorNorm, s16 playerYRot, s16 eyeAtYaw, f32 adjAmt) {
+    f32 tmp;
+    VecSph floorNormSph;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeFIXD3.s")
+    OLib_Vec3fToVecSphGeo(&floorNormSph, floorNorm);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeFIXD4.s")
+    tmp = Math_CosS(floorNormSph.pitch) * Math_CosS(playerYRot - floorNormSph.yaw);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeFIXD0.s")
+    return (Camera_fabsf(tmp) * adjAmt) * Math_CosS(playerYRot - eyeAtYaw);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeSUBJ1.s")
+f32 func_800CD6CC(Actor* actor) {
+    return sqrtf(gTargetRanges[actor->targetMode].rangeSq / gTargetRanges[actor->targetMode].leashScale);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_ModeSUBJ2.s")
+/**
+ * Calculates new at vector for the camera pointing in `eyeAtDir`
+ */
+s32 Camera_CalcAtDefault(Camera* camera, VecSph* eyeAtDir, f32 yOffset, s16 calcSlope) {
```

Comment:
Another option for these awkward functions with `at` in the name is to use `AtVec` in the name, so we signal to the reader we aren't using at as an adverb.

Reply:
I agree these collection of `Camera_CalcAt` is worded weiredly. I wonder if this is a case where we could mix `at` and `lookAt` a little, so name these set of functions `Camera_CalcLookAtXXX` as another idea

Reply:
I'd be OK with `LookAt` replacing `At`

Reply:
I'm kinda restating the same point in different comments, but I'm leaning towards `Camera_CalcLookAtXXX` back to `Camera_CalcAt` for consistency, and I would prefer a global change of `at` -> `lookAt` if that's what we wanted to go for. Open to discussion though, but my preference is just `at` for now